src/main/java/com/aws/greengrass/secretmanager/AWSSecretClient.java

```diff
@@ -53,30 +55,24 @@ public AWSSecretClient(LazyCredentialProvider credentialProvider, DeviceConfigur
 
     /**
      * Fetch secret from AWS cloud.
+     *
      * @param request AWS request for fetching secret from cloud
      * @return AWS secret response
      * @throws SecretManagerException If there is a problem fetching secret
-     * @throws IOException If there is a network error
      */
-    public GetSecretValueResponse getSecret(GetSecretValueRequest request) throws SecretManagerException, IOException {
+    public GetSecretValueResponse getSecret(GetSecretValueRequest request) throws SecretManagerException {
         String errorMsg = String.format("Exception occurred while fetching secrets "
-                        + "from AWSSecretsManager for secret: %s, version: %s, label: %s",
-                request.secretId(), request.versionId(), request.versionStage());
+                        + "from AWSSecretsManager for secret: %s, version: %s, label: %s", request.secretId(),
+                request.versionId(), request.versionStage());
         try {
             validateInput(request);
-            BaseRetryableAccessor accessor = new BaseRetryableAccessor();
-            CrashableSupplier<GetSecretValueResponse, IOException> getSecretValueResponse =
-                    () -> secretsManagerClient.getSecretValue(request);
-            GetSecretValueResponse response = accessor.retry(RETRY_COUNT, BACKOFF_MILLIS, getSecretValueResponse,
-                    new HashSet<>(Collections.singletonList(IOException.class)));
+            GetSecretValueResponse response =
+                    RetryUtils.runWithRetry(retryConfig, () -> secretsManagerClient.getSecretValue(request),
+                            "get-secret-response", logger);
             validateResponse(response);
             return response;
-        } catch (InternalServiceErrorException
-                | DecryptionFailureException
-                | ResourceNotFoundException
-                | InvalidParameterException
-                | InvalidRequestException
-                | IllegalArgumentException e) {
+        } catch (Exception e) {
```

Comment:
also need to handle interrupt exception

Reply:
This method can be called from IPC thread/service pool thread. Since this not being called from a separate thread, not sure if should handle the interrupted exception explicitly.